credential.js

```diff
@@ -26,7 +26,28 @@ var crypto = require('crypto'),
 
   msPerDay = 24 * 60 * 60 * 1000,
   msPerYear = 366 * msPerDay,
-  y2k = new Date(2000, 0, 1),
+  y2k = new Date(2000, 0, 1);
+
+module.exports = function credential(options){
+
+  var defaultOptions = {
```

Comment:
Could this be moved up a scope? Outside the factory function? I don't see a particular reason to reallocate this object every time.
Unless it was put in as the first argument to `mixIn` further down instead to avoid that empty object (seems a idiomatic anyway).